pinot-spi/src/main/java/org/apache/pinot/spi/filesystem/PinotFS.java

```diff
@@ -103,8 +106,13 @@ public boolean move(URI srcUri, URI dstUri, boolean overwrite)
       }
     } else {
       // ensures the parent path of dst exists.
-      URI parentUri = Paths.get(dstUri).getParent().toUri();
-      mkdir(parentUri);
+      try {
+        Path parentPath = Paths.get(dstUri.getPath()).getParent();
```

Comment:
Can you add a test for testing the case when the scheme doesn't match then throw the exception?

Reply:
added a test to cover this case to `LocalPinotFSTest`